### PR TITLE
DHIS2-2440

### PR DESCRIPTION
**Pull request by stian-sandvold**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/dataelement/hibernate/DataElement.hbm.xml [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
     <property name="style" type="jbObjectStyle" column="style" />
 
+    <property name="renderType" type="jbRenderType" column="renderstyle" />
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/dataelement/hibernate/DataElement.hbm.xml]: renderType

File: dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/StaticRenderingConfigurationController.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.util.Set;
+
+@Controller
+@RequestMapping( value = "/staticConfiguration/" )
+@ApiVersion( DhisApiVersion.DEFAULT )
+public class StaticRenderingConfigurationController
+{
+
+    @RequestMapping( value = "renderingOptions", method = RequestMethod.GET )
```
Comment (mortenoh) [inline dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/StaticRenderingConfigurationController.java]: @GetMapping

File: dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/StaticRenderingConfigurationController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.hisp.dhis.dxf2.configuration.StaticRenderingConfiguration;
+import org.hisp.dhis.webapi.mvc.annotation.ApiVersion;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.Set;
+
+@Controller
```
Comment (mortenoh) [inline dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/StaticRenderingConfigurationController.java]: @RestController

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityAttributeObjectBundleHook.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    public <T extends IdentifiableObject> List<ErrorReport> validate( T object, ObjectBundle bundle )
+    {
+        List<ErrorReport> errorReports = new ArrayList<>();
+
+        if ( object != null && object.getClass().isAssignableFrom( TrackedEntityAttribute.class ) )
+        {
+            TrackedEntityAttribute attr = (TrackedEntityAttribute) object;
+            DeviceRenderTypeMap<ValueTypeRenderingObject> map = attr.getRenderType();
+
+            if ( map != null )
```
Comment (mortenoh) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityAttributeObjectBundleHook.java]: Maybe you can just check map == null here and break out? so we don't have another level of indent

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityAttributeObjectBundleHook.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
+package org.hisp.dhis.dxf2.metadata.objectbundle.hooks;
```
Comment (mortenoh) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityAttributeObjectBundleHook.java]: copyright

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/DataElementObjectBundleHook.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package org.hisp.dhis.dxf2.metadata.objectbundle.hooks;
```
Comment (mortenoh) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/DataElementObjectBundleHook.java]: copyright

**Pull request merged**